### ISSUE TITLE
Update grpcio to version >= 1.53.0

### ISSUE THREAD

**Issue by nico-stefani** [wazuh/wazuh#18215]
|Wazuh version|Component|
|---|---|
| 4.7.0 | API/Framework |

## Description

Due to a vulnerability detected in `grrpcio` library prior to `1.38.1`, it needs to be updated at least to version `1.53.0` or later.

Scan result:

```
{
    "package_name": "grpcio",
    "package_version": "1.38.1",
    "package_affected_version": "<1.53.0",
    "vuln_description": "Grpcio 1.53.0 includes a fix for a Connection Termination vulnerability. The prior versions contain a vulnerability whereby a client can cause a termination of connection between a HTTP2 proxy and a gRPC server: a base64 encoding error for -bin suffixed headers will result in a disconnection by the gRPC server, but is typically allowed by HTTP2 proxies.\r\nhttps://github.com/advisories/GHSA-9hxf-ppjv-w6rq",
    "safety_id": "59868"
},
{
    "package_name": "grpcio",
    "package_version": "1.38.1",
    "package_affected_version": "<1.53.0",
    "vuln_description": "Grpcio 1.53.0 includes a fix for a Connection Confusion vulnerability. When gRPC HTTP2 stack raised a header size exceeded error, it skipped parsing the rest of the HPACK frame. This caused any HPACK table mutations to also be skipped, resulting in a desynchronization of HPACK tables between sender and receiver. If leveraged, say, between a proxy and a backend, this could lead to requests from the proxy being interpreted as containing headers from different proxy clients - leading to an information leak that can be used for privilege escalation or data exfiltration.\r\nhttps://github.com/advisories/GHSA-cfgp-2977-2fmm",
    "safety_id": "59869"
},
{
    "package_name": "grpcio",
    "package_version": "1.38.1",
    "package_affected_version": "<1.53.0",
    "vuln_description": "Grpcio 1.53.0 includes a fix for a Reachable Assertion vulnerability. \r\nhttps://github.com/advisories/GHSA-6628-q6j9-w8vg",
    "safety_id": "59867"
}
```
- [Grpcio versions changelog](https://github.com/grpc/grpc/releases)

## Checks

The following elements have been updated or reviewed (should also be checked if no modification is required):
- [ ] Tests (unit tests, API integration tests).
- [ ] Changelog.
- [ ] Documentation.
- [ ] Integration test mapping (using `api/test/integration/mapping/_test_mapping.py`).

**Comment by Selutario** [comment]
According to NVD:
- [CVE-2023-1428](https://nvd.nist.gov/vuln/detail/CVE-2023-1428): From (including) 1.51.0 to (excluding) 1.55.0.
- [CVE-2023-32731](https://nvd.nist.gov/vuln/detail/CVE-2023-32731): From (including) 1.53.0 to (excluding) 1.55.0.
- [CVE-2023-32732](https://nvd.nist.gov/vuln/detail/CVE-2023-32732): Up to (excluding) 1.53.0.

Since `grpcio` version in the framework is currently 1.38.1, it's only affected by [CVE-2023-32732](https://nvd.nist.gov/vuln/detail/CVE-2023-32732).

**Comment by EduLeon12** [comment]
## Issue Update

New vulnerability found for grcpio

```
{
            "package_name": "grpcio",
            "package_version": "1.38.1",
            "package_affected_version": "<1.53.2",
            "vuln_description": "Grpcio 1.53.2, 1.54.3, 1.55.3 and 1.56.2 include a fix for CVE-2023-4785: Lack of error handling in the TCP server in Google's gRPC starting version 1.23 on posix-compatible platforms (ex. Linux) allows an attacker to cause a denial of service by initiating a significant number of connections with the server. Note that gRPC C++ Python, and Ruby are affected, but gRPC Java, and Go are NOT affected.\r\nhttps://github.com/grpc/grpc/pull/33656",
            "safety_id": "61191"
        }
```

Grcpcio upgrade needs to be extended to version  >=1.53.2 to include a fix for `CVE-2023-4785`.

**Comment by fdalmaup** [comment]
# Issue Update

The `grpcio` dependency was updated for Wazuh `4.8.0` in #13265:

https://github.com/wazuh/wazuh/blob/24e541b489e11578e7979bb04ecbf6bc3b5f3497/framework/requirements.txt#L44.

Since there are no available `1.53.0` or higher version wheels for CentOS 6 and the Wazuh `4.7.1` code is soon to be delivered, there is no time to analyze and execute the dependency upgrade for said version.
